### ISSUE TITLE
Potential bug - contract creation data

### ISSUE THREAD

**Issue by ArranKitson** [TokenFlowInsights/EDW-Feedback#9]
Two potential bugs reported in discord:

https://discord.com/channels/806456638483595265/946097180924604496/1058180561098776576

https://discord.com/channels/806456638483595265/946097180924604496/1058187515342888990

**Comment by kchojn** [comment]
- Problem: *I would expect a row in state_diffs for reason = 'contract creation'*
    ```sql
    select * from edw.raw.calls where tx_hash = '0x959a36e5773c5d68477f95e8c41254d9bbbdae857ba66ef6fdfab23e8f77cc3b' and block >= 16010522 and block <= 16010523
    
    select * from edw.raw.state_diffs where tx_hash = '0x959a36e5773c5d68477f95e8c41254d9bbbdae857ba66ef6fdfab23e8f77cc3b' and block >= 16010522 and block <= 16010523
    ```
    
    We have a call `create` that has the initialization code (run only once). This code doesn't return any value (this is deployment code, but it is not equivalent to deployed code), so no new value is pushed into the state tree.`codeHash` doesn't change (no state change), hence there is no `contract creation` in the state, but we have the nonce change (0 -> 1).
    
- Summary: The expected `contract creation` in the state should not be there, due to the fact that the new code is not pushed, so the state doesn't change (no `codeHash` change).

**Comment by kchojn** [comment]
- Problem: *there is a contract creation not marked as call_type = 'create' (marked as "call") in calls table, but it is marked as a 'contract creation' state diff*
  ```sql
  select *  from edw.raw.state_diffs where reason='contract creation' and tx_hash = '0xba873263354b8fe4247a6bce5c7ae41d908b1b4c610d181cb0774c6f6389484d' and status = true and block >= 121816 and block <= 121817
  
  select *  from edw.raw.calls where tx_hash = '0xba873263354b8fe4247a6bce5c7ae41d908b1b4c610d181cb0774c6f6389484d' and status = true and block >= 121816 and block <= 121817
  ```
  
  Call has the wrong type, it's expected as `created`, while we have a normal `call`. There is a state change and a contract creation, so it looks like the call has the wrong type. This is one of the first blocks, so it is possible that the problem was with the ETL.

- Summary: The call has the wrong type. We reprocess the whole history and in the new release - it will have the correct one.

**Comment by garythung** [comment]
Thanks Karol this is great!

**Comment by garythung** [comment]
Any progress on this?

**Comment by kchojn** [comment]
Yes, we are reprocessing the history, in about ~1-1.5 months we are doing a new `EDW` release (with fixes and many features)
